x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils/import_rules_utils.ts

```diff
@@ -0,0 +1,396 @@
+                    to,
+                    type,
+                    threat,
+                    threshold,
+                    threatFilters,
+                    threatIndex,
+                    threatIndicatorPath,
+                    threatQuery,
+                    threatMapping,
+                    threatLanguage,
+                    concurrentSearches,
+                    itemsPerSearch,
+                    references,
+                    note,
+                    version,
+                    exceptionsList: [...exceptions],
+                    anomalyThreshold,
+                    machineLearningJobId,
+                    actions,
+                  });
+                  resolve({
+                    rule_id: ruleId,
+                    status_code: 200,
+                  });
+                } else if (rule != null) {
+                  resolve(
+                    createBulkErrorObject({
+                      ruleId,
+                      statusCode: 409,
+                      message: `rule_id: "${ruleId}" already exists`,
+                    })
+                  );
+                }
+              } catch (err) {
+                resolve(
+                  createBulkErrorObject({
+                    ruleId,
+                    statusCode: err.statusCode ?? 400,
+                    message: err.message,
+                  })
+                );
+              }
+            } catch (error) {
+              reject(error);
+            }
+          });
+          return [...accum, importsWorkerPromise];
+        }, [])
+      );
+      importRuleResponse = [...importRuleResponse, ...newImportRuleResponse];
+    }
+
+    return importRuleResponse;
+  }
+};
+
+// TODO: Batch this upfront and send down to check against
+export const checkExceptions = async ({
+  ruleId,
+  exceptions,
+  exceptionsClient,
+}: {
+  ruleId: string;
+  exceptions: ListArray;
+  exceptionsClient: ExceptionListClient | undefined;
+}): Promise<[BulkError[], ListArray]> => {
+  let ruleExceptions: ListArray = [];
+  let errors: BulkError[] = [];
+
+  if (!exceptions.length || exceptionsClient == null) {
+    return [[], exceptions];
+  }
+  for await (const exception of exceptions) {
+    const { list_id: listId, namespace_type: namespaceType } = exception;
+    const list = await exceptionsClient.getExceptionList({
+      id: undefined,
+      listId,
+      namespaceType,
+    });
+
+    if (list != null) {
+      ruleExceptions = [...ruleExceptions, { ...exception, id: list.id }];
+    } else {
+      // if exception is not found remove link
+      errors = [
+        ...errors,
+        createBulkErrorObject({
+          ruleId,
+          statusCode: 400,
+          message: `Rule with rule_id: "${ruleId}" references a non existent exception list of list_id: "${listId}". Reference has been removed.`,
+        }),
+      ];
+    }
+  }
+
+  return [errors, ruleExceptions];
+};
+
+export const importExceptionsHelper = async ({
```

Comment:
`Helper` is an interesting suffix to use here... I don't think it's adding much and you could probably drop it.

Reply:
Renamed!